### PR TITLE
CNDIT 1212: Update Java post-processing for HEPATITIS_DATAMART

### PR DESCRIPTION
**Pull request by sveselev**
## Description
This PR includes implementation for processing Investigation post-processing result to produce and consume Datamart related to call `sp_hepatitis_datamart_postprocessing` stored procedure.

- **ProcessDatamartData.java**: Datamart topic creator, transforms the data from Investigation post-process result into the topic
- PostProcessingService.java: added Datamart data cache and scheduled task method to process it and call `sp_hepatitis_datamart_postprocessing` 

#JIRA
[CNDIT-1212](https://cdc-nbs.atlassian.net/browse/CNDIT-1212)

### CODE REVIEW

File: post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java [hunk 1]
```diff
@@ -63,7 +69,25 @@ public void postProcessMessage(
...
+    public void postProcessDatamart(
+            @Header(KafkaHeaders.RECEIVED_TOPIC) String topic,
+            @Payload String payload) {
+        try {
+            JsonNode payloadNode = objectMapper.readTree(payload);
+            logger.info("Got this payload: {} from the topic: {}", payloadNode, topic);
+
+            Datamart dmData = objectMapper.readValue(payloadNode.get(PAYLOAD).toString(), Datamart.class);
+            Map<Long, Long> dmMap = new HashMap<>();
+            dmMap.put(dmData.getPublicHealthCaseUid(), dmData.getPatientUid());
```
Comment (upasanapattnaik-eq) [inline post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java:82]: Currently, we do not monitor PatientUid being 1, correct?

File: post-processing-service/src/main/resources/application.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ spring:
       patient: nrt_patient
       provider: nrt_provider
       notification: nrt_notifications
+      datamart: nrt_datamart
```
[the remark this answers is not in the document]
Comment (sveselev) [reply post-processing-service/src/main/resources/application.yaml:10]: @upasanapattnaik-eq No, we're not sinking it, it's better to rename it.

File: post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java (hunk 1 above)
Comment (sveselev) [reply post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java:82]: @upasanapattnaik-eq For now there is no restriction, need to be added.

**Review by upasanapattnaik-eq — CHANGES_REQUESTED** [review]
Update to topic name for datamart

**Review by upasanapattnaik-eq — APPROVED** [review]
LGTM! Thank you so much for implementing 👍

**Pull request merged**